### PR TITLE
Kafka consumer cli

### PR DESCRIPTION
**Pull request by itkovian**

### CODE REVIEW

File: vsc-ci.ini [hunk 1]
```diff
@@ -1,3 +1,4 @@
 [vsc-ci]
 pip3_install_tox=1
 py3_tests_must_pass=1
+py3_only=1
```
Comment (wdpypere) [inline vsc-ci.ini:4]: euh, no. still in use on centos 7 machines.

**Review by wdpypere — CHANGES_REQUESTED** [review]

File: lib/vsc/utils/cache.py [hunk 2]
```diff
@@ -158,19 +158,17 @@ def close(self):
...
-            g = gzip.GzipFile(mode='wb', fileobj=f)
-            pickled = jsonpickle.encode(self.new_shelf)
-            # .encode() is required in Python 3, since we need to pass a bytestring
-            g.write(pickled.encode())
-            g.close()
-            f.close()
-
-            self.log.info('closing the file cache at %s', self.filename)
+        with open(self.filename, 'wb') as f:
+            if not f:
```
Comment (wdpypere) [inline lib/vsc/utils/cache.py:162]: will this ever be not true? Also, I don't know if I like 1 letter variables :smile:

File: lib/vsc/utils/cli.py [hunk 3]
```diff
@@ -0,0 +1,193 @@
+#
+# Copyright 2020-2023 Ghent University
```
Comment (wdpypere) [inline lib/vsc/utils/cli.py:2]: how can this be true on a new file?

File: lib/vsc/utils/cli.py [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+    KAFKA_COMMON_OPTIONS = {
+        'topic': ("Kafka topics to produce/consume", None, "store", "xdmod"),
+        'brokers': ("List of kafka brokers, comma separated", "strlist", "store", None),
+        'security_protocol': ("Security protocol to use, e.g., SASL_SSL", str, "store", "PLAINTEXT"),
+        'ssl': ("Comma-separated key=value list of SSL options for underlying kafka lib", "strlist", "store", []),
+        'sasl': ("Comma-separated key=value list of SASL options for the underlying kafka lib", "strlist", "store", []),
+        # Very advanced/dangerous usage:
+        # e.g. on initial usage of consumer, pass --kafka=auto_offset_reset=earliest
+        #    to start from the earliest offset in abscence of (first) commit
+        'kafka': ("Comma-separated key=value list of allowed options for the underlying kafka lib",
```
Comment (wdpypere) [inline lib/vsc/utils/cli.py:80]: if this is dangerous, maybe put an extra option so people have to use 2 options, or put the word `dangerous` in the description.

File: lib/vsc/utils/consumer.py [hunk 5]
```diff
@@ -0,0 +1,176 @@
+#
+# Copyright 2020-2023 Ghent University
```
Comment (wdpypere) [inline lib/vsc/utils/consumer.py:2]: 2023-2023

File: test/cli.py [hunk 6]
```diff
@@ -0,0 +1,46 @@
+#
+# Copyright 2021-2023 Ghent University
```
Comment (wdpypere) [inline test/cli.py:2]: 2023-2023

File: test/consumer.py [hunk 7]
```diff
@@ -0,0 +1,177 @@
+#
+# Copyright 2021-2023 Ghent University
```
Comment (wdpypere) [inline test/consumer.py:2]: here as well

File: Jenkinsfile [hunk 8]
```diff
@@ -9,7 +9,6 @@ node {
         sh 'git clean -fxd'
     }
     stage('test') {
-        sh 'python2.7 -V'
```
Comment (wdpypere) [inline Jenkinsfile:12]: can't be py3_only yet. vsc-utils is still in use on centos 7 machines.

File: lib/vsc/utils/cli.py (hunk 3 above)
Comment (itkovian) [reply lib/vsc/utils/cli.py:2]: The file comes from elsewhere.

File: lib/vsc/utils/cli.py (hunk 4 above)
Comment (itkovian) [reply lib/vsc/utils/cli.py:80]: I think Stijn added that comment. Normally one should not know things that are dangerous. in any case, we use it all the time to pass the credentials and stuff.

File: lib/vsc/utils/consumer.py (hunk 5 above)
Comment (itkovian) [reply lib/vsc/utils/consumer.py:2]: No, code comes from elsewhere.

File: Jenkinsfile (hunk 8 above)
Comment (itkovian) [reply Jenkinsfile:12]: Can we version lock it there?

Comment (wdpypere) [reply Jenkinsfile:12]: as long as it is in use on python2.7 I can not agree to making it py3 only.

File: setup.py [hunk 9]
```diff
@@ -40,6 +40,7 @@
     'lockfile >= 0.9.1',
     'netifaces',
     'jsonpickle',
+    'kafka',
```
Comment (wdpypere) [inline setup.py:43]: does this mean we will install kafka on all machines?

**Comment by itkovian** [comment]
Moving code to vsc-kafka

**Pull request closed without merging**